x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -184,59 +185,98 @@ export class ExecutionHandler<
-              kibanaBaseUrl: this.taskRunnerContext.kibanaBaseUrl,
-              alertParams: this.rule.params,
-              actionParams: action.params,
-              ruleUrl: this.buildRuleUrl(spaceId),
+              actionTypeId,
+              actionParams: transformActionParams({
+                actionsPlugin,
+                alertId: ruleId,
+                alertType: this.ruleType.id,
+                actionTypeId,
+                alertName: this.rule.name,
+                spaceId,
+                tags: this.rule.tags,
+                alertInstanceId: alert.getId(),
+                alertActionGroup: actionGroup,
+                alertActionGroupName: this.ruleTypeActionGroups!.get(actionGroup)!,
+                context: alert.getContext(),
+                actionId: action.id,
+                state: alert.getScheduledActionOptions()?.state || {},
+                kibanaBaseUrl: this.taskRunnerContext.kibanaBaseUrl,
+                alertParams: this.rule.params,
+                actionParams: action.params,
+                ruleUrl: this.buildRuleUrl(spaceId),
+              }),
             }),
-          }),
-        };
-
-        await this.actionRunOrAddToBulk({
-          enqueueOptions: this.getEnqueueOptions(actionToRun),
-          bulkActions,
-        });
-
-        logActions.push({
-          id: action.id,
-          typeId: action.actionTypeId,
-          alertId: alert.getId(),
-          alertGroup: action.group,
-        });
-
-        if (this.isRecoveredAlert(actionGroup)) {
-          alert.scheduleActions(action.group as ActionGroupIds);
-        } else {
-          if (action.frequency?.throttle) {
-            alert.updateLastScheduledActions(
-              action.group as ActionGroupIds,
-              this.generateActionHash(action)
-            );
+          };
+
+          await this.actionRunOrAddToBulk({
+            enqueueOptions: this.getEnqueueOptions(actionToRun),
+            bulkActions,
+          });
+
+          logActions.push({
+            id: action.id,
+            typeId: action.actionTypeId,
+            alertId: alert.getId(),
+            alertGroup: action.group,
+          });
+
+          if (this.isRecoveredAlert(actionGroup)) {
+            alert.scheduleActions(action.group as ActionGroupIds);
           } else {
-            alert.updateLastScheduledActions(action.group as ActionGroupIds);
+            if (this.isActionOnThrottleInterval(action)) {
+              alert.updateLastScheduledActions(
+                action.group as ActionGroupIds,
+                this.generateActionHash(action)
+              );
+            } else {
+              alert.updateLastScheduledActions(action.group as ActionGroupIds);
+            }
+            alert.unscheduleActions();
+          }
+        } else {
+          const summarizedAlerts = await this.getSummarizedAlerts({ action, spaceId, ruleId });
+          const actionToRun = {
```

Comment:
**question:** should we skip the action if there isn't any activity to summarize?

```suggestion
          const summarizedAlerts = await this.getSummarizedAlerts({ action, spaceId, ruleId });
          if (summarizedAlerts.all.count === 0) { continue; }
          const actionToRun = {
```

Reply:
I am also not sure, since this is a summary user may want to be notified even if there isn't any alert..
I think `ruleType.getSummarizedAlerts` returns `{new: {count:0, alerts:[]}} in that case.`
@ymao1 WDYT?

Reply:
cc @shanisagiv1 for input. Should we send empty summaries where there isn't activity? (no alerts new, active nor recovered).

Reply:
When the summarization is defined to be sent per rule run, we won't send a summary when no alerts are detected by the rule right?
For summarization per time interval, it feels to me like it's a bit different since the user expects to get an email every x hours. So sending an email that said "everything is fine, no alerts are detected over x hours" will be more accurate. wdyt?

Reply:
Makes sense to me. @ymao1 @ersin-erdal WDYT?

Reply:
That makes sense!

Reply:
Fixed with: https://github.com/elastic/kibana/pull/147360/commits/25213b66ea25305a55d7b2ae5dfb79ef16f9be5b

Skips scheduling an action for `summary per rule run` when there isn't any alert. Checks it even before making getSummarizedAlerts request.

For `summary on custom interval`, message can be modified by using `if` in mustashe template.

Reply:
Just another data point, for detection rules, they do not send an action where there are no alerts returned in their hourly/daily/weekly rollups.

Reply:
cc @shanisagiv1 if that ^^ changes things. We'll follow through what you originally recommended but if what @ymao1 mentioned changes your mind, we can do a follow up to make the change.